server/util/map.js

```diff
@@ -157,6 +158,54 @@ async function getTrafficByLocation(coordinates) {
 	return traffic;
 }
 
+async function createCrime(crimeList) {
+	const createdCrimes = [];
+
+	for (const crime of crimeList) {
+		const createdCrime = new Crime({
+			type: crime.type,
+			address: crime.address,
+			date: crime.date,
+			location: {
+				coordinates: [crime.location.longtitude, crime.location.latitude],
+			},
+		});
+
+		try {
+			const sess = await mongoose.startSession();
+			sess.startTransaction();
+			await createdCrime.save({ session: sess });
+			await sess.commitTransaction();
```

Comment:
I wonder why we are using transaction here? I thought that if we do, shouldn't the transaction be outside of for loop. Usually, a transaction is used if we have multiple actions, and we want they all succeed or all fail. Here your session has only one action before you commit